pywbem_mock/_dmtf_cim_schema.py

```diff
@@ -0,0 +1,427 @@
+        Raises:
+            ValueError: if the schema cannot be retrieved
+                from the DMTF web site.
+            TypeError:  if the `schema_version` is not
+                a valid tuple with 3 integer components
+
+        """
+        if not isinstance(schema_version, tuple):
+            raise TypeError('schema_version must be tuple not %s' %
+                            schema_version)
+        if len(schema_version) != 3:
+            raise TypeError('DMTF Schema must have 3 integer components, '
+                            '(major, minor, update) not %s' % schema_version)
+        schema_type = 'Experimental' if use_experimental else 'Final'
+        self._schema_version = schema_version
+        self._schema_root_dir = schema_root_dir
+
+        mof_dir = 'mof%s%s' % (schema_type, self.schema_version_str)
+        self._schema_mof_dir = os.path.join(self._schema_root_dir, mof_dir)
+
+        self._dmtf_schema_version = 'cim_schema_%s' % (self.schema_version_str)
+        self._mof_zip_bn = '%s%s-MOFs.zip' % (self._dmtf_schema_version,
+                                              schema_type)
+        self._mof_zip_url = \
+            'http://www.dmtf.org/standards/cim/cim_schema_v%s%s%s/%s' % \
+            (schema_version[0], schema_version[1], schema_version[2],
+             self._mof_zip_bn)
+        schema_mof_bld_name = self._dmtf_schema_version + '.mof'
+
+        self._mof_zip_filename = os.path.join(self.schema_root_dir,
+                                              self._mof_zip_bn)
+        self._schema_mof_filename = os.path.join(self.schema_mof_dir,
+                                                 schema_mof_bld_name)
+
+        self.verbose = verbose
+
+        # Setup the local copy of the DMTF CIM schema which may include creating
+        # the schema directory, downloading the schema zip file from the DMTF
+        # web site, and expanding the schema into a subdirectory of the schema
+        # directory.
+        self._setup_dmtf_schema()
+
+    @property
+    def schema_version(self):
+        """
+        :func:`py:tuple` of 3 integers with major version, minor version, and
+        revision of the DMTF Schema installed. Ex (2, 49, 0) defines DMTF
+        schema version 2.49.0.
+        """
+        return self._schema_version
+
+    @property
+    def schema_version_str(self):
+        """
+        :term:`string` with the DMTF CIM schema version in the form
+        <major version>.<minor version>.<revison>.
+        """
+        return '%s.%s.%s' % (self.schema_version[0], self.schema_version[1],
+                             self.schema_version[2])
+
+    @property
+    def schema_root_dir(self):
+        """
+        :term:`string` the directory in which the DMTF CIM Schema
+        MOF is downloaded and expanded.
+        """
+        return self._schema_root_dir
+
+    @property
+    def schema_mof_dir(self):
+        """
+        :term:`string` the directory in which the DMTF CIM Schema
```

Comment:
It does not become clear from the description whether this is the complete (relative?) path name of the directory, or just the one directory below schema_dir.
Suggestion:

    :term:`string`: Path name of the directory in which the DMTF CIM Schema ...

Reply:
Good catch. DONE